src/mlpack/methods/ann/activation_functions/gelu_function.hpp

```diff
@@ -0,0 +1,94 @@
+
+/**
+ * @file gelu_function.hpp
+ * @author Himanshu Pathak
+ *
+ * Definition and implementation of the Gaussian Error Linear Unit (GELU)
+ * function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_GELU_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_GELU_FUNCTION_HPP
+# define PI 3.141592653589793238462643383279502884L /* pi */
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The GELU function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = 0.5 * x * {1 + tanh[(2/pi)^(1/2) * (x + 0.044715 * x^3)]} \\
+ * f'(x) = 0.5 * tanh(0.0356774 * x^3) + 0.797885 * x) + 
+ *         (0.0535161x^3 + 0.398942 * x) * 
+ *         sech^2(0.0356774 * x^3+0.797885 * x) + 0.5\\
+ * @f}
+ */
+class GELUFunction
+{
+ public:
+  /**
+   * Computes the GELU function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x)
+  {
+    return 0.5 * x * (1 + std::tanh(std::sqrt(2 / PI) *
```

Comment:
Looking at the paper, I'm wondering if we should implement the simplified approximation instead, sounds like they used that form for all experiments. Let me know what you think.

Reply:
@zoq 

![a](https://user-images.githubusercontent.com/37804981/74088043-00700280-4ab8-11ea-86d8-9cf524cdc7b3.jpeg)

On reading paper I found out that they are using 0.5x(1 + tanh[ (2/π)^(1/2)(x + 0.044715x 3 )]) approximation. I also used the same . Let me know if I am missing something.

Reply:
Ahh, thanks for the clarification, yeah makes sense.